Database xp implementation

Tromodolo opened the pull request.

Tromodolo committed:
Update types to add Usertimer for xp

Tromodolo committed:
Update packages

Tromodolo committed:
Add db.js

Tromodolo committed:
Add xp handling system using db

This doesn't actually work

Tromodolo committed:
Somewhat fixed xp system

Tromodolo commented:
It currently doesn't register whether or not a user has leveled up. Also considering moving stuff like this over to a web api instead of having it native in the bot. Would clean up code and make it easier to manage

Commit:
added in depth JSDoc comments for the types

Added JSDoc comments for Command, and CommandModule Interfaces. UserTimer needs to be fixed up though.

Commit:
fix JSDoc comments, and add to handleExperience()

fixed JSDoc comments on current comments, and added JSDoc comments to the handleExperience() function

miniaxolotl commented:
added some JSDoc comments need to finish the comment on UserTimer here https://github.com/Tromodolo/Kurisu-Node/blob/135157514e5a7de9be792a394d37af7b07016dec/types.ts I didn't know what it did @Tromodolo

Tromodolo committed:
Fix return value for command typing

Tromodolo committed:
Add express backend subfolder

This won't run when the bot runs, but I don't know if this should get it's own repo or not

Tromodolo committed:
Add backend to debugging in vs code

Tromodolo committed:
Update backend to have an api endpoint for user experience

Endpoint is /api/user/experience

Tromodolo committed:
Update packages to include backend

Tromodolo committed:
Update config example to match new api key setup

Tromodolo committed:
Move backend to its own repo

Tromodolo committed:
Add api endpoint property to the config

Tromodolo committed:
Remove backend packages from package.json and package-lock

Tromodolo committed:
Change handleExperience to move from using db directly to using web api

Tromodolo committed:
Fix UserTimer typings
